### DISCUSSION TITLE
Load extensions from node_modules

### DISCUSSION THREAD

**Discussion by dimitarkolev** [directus/directus#12408] [Archive]
How to make directus load extension that is installed as a npm module? I the documentation its only written how to develop an extension and put the output js inside the extensions folder but not how to do same thing with npm.

**Comment by rijkvanzanten** [comment]
@nickrum ?

**Comment by nickrum** [comment]
Directus will load any extension installed by npm that uses the naming pattern `directus-extension-<custom-name>` or `@<custom-scope>/directus-extension-<custom-name>`.

We intentionally chose not to document this while we were in RC phase. Now that the extension API has been stabilized, it's about time to document npm extensions 👍

**Reply by rijkvanzanten** [reply]
(cc @erondpowell)

**Comment by hairmare** [comment]
As part of evaluating if directus is going to be my no-code platform of choice, I built an interface extension that is published to npm via github actions and can confirm that publishing it with a `directus-extension-` prefix works as advertised.

The extension is here: https://github.com/hairmare/directus-extension-interface-recurringrule. The bits related to packaging are in `.github/workflows/ci.yaml` and `.npmignore` if anyone wants to have a closer look.

The extension itself still has a long way to go (i'm not that much of a frontend dev and am learning ts while i build it).
